core/trino-main/src/test/java/io/trino/sql/TestExpressionInterpreter.java

```diff
@@ -1397,6 +1421,61 @@ public void testLikeOptimization()
         assertOptimizedEquals("unbound_string LIKE unbound_pattern ESCAPE unbound_string", "unbound_string LIKE unbound_pattern ESCAPE unbound_string");
     }
 
+    @Test
+    public void testLikeCharOptimization()
+    {
+        // constant literal pattern of length shorter than value length
+        assertOptimizedEquals("unbound_char LIKE 'abc'", "false");
+        assertOptimizedEquals("unbound_char LIKE 'abc' ESCAPE '#'", "false");
+        assertOptimizedEquals("unbound_char LIKE 'ab#_' ESCAPE '#'", "false");
+        assertOptimizedEquals("unbound_char LIKE 'ab#%' ESCAPE '#'", "false");
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'abc'", "false");
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'abc' ESCAPE '#'", "false");
+
+        // constant non-literal pattern of length shorter than value length
+        assertOptimizedEquals("unbound_char LIKE 'ab_'", "unbound_char LIKE 'ab_'");
+        assertOptimizedEquals("unbound_char LIKE 'ab%'", "unbound_char LIKE 'ab%'");
+        assertOptimizedEquals("unbound_char LIKE 'ab%' ESCAPE '#'", "unbound_char LIKE 'ab%' ESCAPE '#'");
+
+        // constant literal pattern of length equal to value length
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'abcd'", "CAST(unbound_char AS char(4)) = CAST('abcd' AS char(4))");
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'abcd' ESCAPE '#'", "CAST(unbound_char AS char(4)) = CAST('abcd' AS char(4))");
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'jaźń'", "CAST(unbound_char AS char(4)) = CAST('jaźń' AS char(4))");
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'ab#_' ESCAPE '#'", "false");
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'ab#%' ESCAPE '#'", "false");
+
+        // constant non-literal pattern of length equal to value length
+        assertOptimizedEquals("CAST(unbound_char AS char(4)) LIKE 'ab#%' ESCAPE '\\'", "CAST(unbound_char AS char(4)) LIKE 'ab#%' ESCAPE '\\'");
+
+        // constant pattern of length longer than value length
```

Comment:
nit: constant literal pattern.

Or just `literal pattern` and drop `constant` from all the comments.

Reply:
"constant" and "literal" can both be understood as meaning "same for all rows" or "containing no wildcards"

i am using both for disambiguation